mod_ci/controllers.py

```diff
@@ -463,23 +463,45 @@ def queue_test(db, gh_commit, commit, test_type, branch="master", pr_nr=0):
     """
     from run import log
 
-    fork = Fork.query.filter(Fork.github.like("%/CCExtractor/ccextractor.git")).first()
+    fork = Fork.query.filter(Fork.github.like(
+        "%/CCExtractor/ccextractor.git")).first()
+    fork_id = Test.query.filter(Test.fork_id == Fork.id).first()
+    githubapi = requests.get(
+        "https://api.github.com/repos/ccextractor/ccextractor/pulls/{}".format())
+    userjson = json.loads(githubapi.body)
+    user = userjson['user']['id']
+
+    record_from_db = BlockedUsers.query.filter_by(userID=user).first()
+    blocked = False
 
-    if test_type == TestType.pull_request:
-        branch = "pull_request"
+    if user == record_from_db.userID:
+        log.critical("Error. User in Blacklist!")
+        blocked = True
+    else:
+        if test_type == TestType.pull_request:
+            branch = "pull_request"
 
-    linux_test = Test(TestPlatform.linux, test_type, fork.id, branch, commit, pr_nr)
-    db.add(linux_test)
-    windows_test = Test(TestPlatform.windows, test_type, fork.id, branch, commit, pr_nr)
-    db.add(windows_test)
-    db.commit()
+        linux_test = Test(TestPlatform.linux, test_type, fork.id, branch, commit, pr_nr)
+        db.add(linux_test)
+        windows_test = Test(TestPlatform.windows, test_type, fork.id, branch, commit, pr_nr)
+        db.add(windows_test)
+        db.commit()
 
-    # Update statuses on GitHub
-    if gh_commit is not None:
-        status_entries = {
-            linux_test.platform.value: linux_test.id,
-            windows_test.platform.value: windows_test.id
-        }
+        # Update statuses on GitHub
+        if gh_commit is not None:
+            status_entries = {
+                linux_test.platform.value: linux_test.id,
+                windows_test.platform.value: windows_test.id
+            }
+
+    if blocked is True:
+        gh_commit.post(
+            state=Status.ERROR,
+            description="Tests cancelled! You may be blocked.",
+            context="CI - {name}".format(name=platform_name),
+            target_url=url_for('test.by_id', test_id=test_id, _external=True)
+        )
+    else:
```

Comment:
This if-else makes no sense. Why not return early when there's a match for a blocked user, so that the remaining flow can be left untouched?